### PR TITLE
Preferences window

### PR DESCRIPTION
**Pull request by Markvanderuit**
Application now has small preferences window for future use. Allows you to set application color for now, although that is not stored in config yet.

### CODE REVIEW

**Comment by Markvanderuit** [comment]
Apologies for the failing builds, but Github died for over 6 hours today, and checks started appearing only hours after.

File: src/main/java/gui/misc/TweakingHelper.java [hunk 1]
```diff
@@ -36,31 +52,105 @@
...
+     */
 
-    // general layout variables used throughout application.
-    public static final int GENERAL_SIZE = 10000;
-    public static final int GENERAL_SPACING = 10;
-    public static final int GENERAL_PADDING = 20;
+    /**
+     * @return primary color.
+     */
+    public static Color getPrimaryColor() {
```
Comment (bart-de-jonge) [inline src/main/java/gui/misc/TweakingHelper.java]: Could you refactor this into something like getNthColor(int n) { return colors[colorChoice][n-1];}?
This is a lot of pretty similar code

File: src/main/java/gui/misc/TweakingHelper.java [hunk 2]
```diff
@@ -36,31 +52,105 @@
...
+     * @return quadratory color.
+     */
+    public static Color getQuadratoryColor() {
+        return colors[colorChoice][3];
+    }
+
+    /**
+     * @return primary color as string.
+     */
+    public static String getPrimaryString() {
```
Comment (bart-de-jonge) [inline src/main/java/gui/misc/TweakingHelper.java]: Same holds for these methods

**Comment by bart-de-jonge** [comment]
Added a few comments on duplicate/similar code, please take a look

File: src/main/java/gui/misc/TweakingHelper.java (hunk 1 above)
Comment (Markvanderuit) [inline src/main/java/gui/misc/TweakingHelper.java]: Yes, I can see why that would be better. I'll implement it.

**Comment by Markvanderuit** [comment]
Reopened PR with cleaner TweakingHelper code and controller test.

File: src/test/java/control/PreferencesViewControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+    public void setUp() throws Exception {
+        controllerManager = Mockito.mock(ControllerManager.class);
+        rootPane = Mockito.mock(RootPane.class);
+        scriptingProject = Mockito.mock(ScriptingProject.class);
+        projectController = Mockito.mock(ProjectController.class);
+        stage = Mockito.mock(Stage.class);
+
+        when(controllerManager.getProjectController()).thenReturn(projectController);
+        when(controllerManager.getRootPane()).thenReturn(rootPane);
+        //when(rootPane.getPrimaryStage()).thenReturn(stage);
```
Comment (geenen124) [inline src/test/java/control/PreferencesViewControllerTest.java]: Remove this commented code or explain why it's commented?

File: src/main/java/gui/modal/ReloadModalView.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+    /**
+     * Initialize the view of this modal.
+     */
+    private void initializeView() {
+        this.informationLabel.setText("This will cause the application to restart!\n"
+            + "Do you wish to save project changes?");
+    }
+
+
```
Comment (geenen124) [inline src/main/java/gui/modal/ReloadModalView.java]: Extra lines?

**Comment by geenen124** [comment]
Looks good!

**Pull request merged**